resources/aws_ec2.go

```diff
@@ -486,6 +489,67 @@ func (obj *AwsEc2Res) longpollWatch() error {
 	}
 }
 
+// snsWatch uses amazon cloudwatch events and simple notification service to
+// detect ec2 instance state changes.
+func (obj *AwsEc2Res) snsWatch() error {
+	send := false
+	var exit *error
+	defer obj.wg.Wait()
+	defer close(obj.closeChan)
+	// set up the sns endpoint
+	snsEndpoint := obj.snsServer()
+	// shutdown the sns endpoint when we're done
+	obj.wg.Add(1)
+	go func() {
+		defer obj.wg.Done()
+		select {
+		case <-obj.closeChan:
+			if err := snsEndpoint.Shutdown(nil); err != nil {
+				log.Printf("error stopping sns endpoint: %s", err)
```

Comment:
Is there a better way to handle this error? I can't pass it to awsChan, since the channel's not being read from anymore, and I can't return it because it's in a goroutine...